engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java

```diff
@@ -0,0 +1,142 @@
+    @Override
+    public void initialise() {
+        super.initialise();
+        playerCamera = worldRenderer.getActiveCamera();
+        addDesiredStateChange(new SetViewportToSizeOf(READ_ONLY_GBUFFER));
+        addDesiredStateChange(new EnableMaterial(CHUNK_SHADER.toString()));
+        chunkShader = getMaterial(CHUNK_SHADER);
+    }
+
+    /**
+     * Renders the world's semi-transparent blocks, i.e. tree foliage and terrain plants.
+     * Does not render fully opaque blocks, i.e. the typical landscape blocks.
+     *
+     * Takes advantage of the two methods
+     *
+     * - WorldRenderer.increaseTrianglesCount(int)
+     * - WorldRenderer.increaseNotReadyChunkCount(int)
+     *
+     * to publish some statistics over its own activity.
+     */
+    @Override
+    public void process() {
+        PerformanceMonitor.startActivity("rendering/chunksAlphaReject");
+
+        final Vector3f cameraPosition = playerCamera.getPosition();
+
+        int numberOfRenderedTriangles = 0;
+        int numberOfChunksThatAreNotReadyYet = 0;
+
+        READ_ONLY_GBUFFER.bind();
+        chunkShader.setFloat("clip", 0.0f, true);
+        chunkShader.activateFeature(ShaderProgramFeature.FEATURE_ALPHA_REJECT);
+
+        playerCamera.lookThrough(); // TODO: remove. Placed here to make the dependency explicit.
+
+        while (renderQueues.chunksAlphaReject.size() > 0) {
+            RenderableChunk chunk = renderQueues.chunksAlphaReject.poll();
+
+            if (chunk.hasMesh()) {
+                final Vector3f chunkPosition = chunk.getPosition().toVector3f();
+                final Vector3f chunkPositionRelativeToCamera =
+                        new Vector3f(chunkPosition.x * ChunkConstants.SIZE_X - cameraPosition.x,
+                                chunkPosition.y * ChunkConstants.SIZE_Y - cameraPosition.y,
+                                chunkPosition.z * ChunkConstants.SIZE_Z - cameraPosition.z);
+
+                chunkShader.setFloat3("chunkPositionWorld",
+                        chunkPosition.x * ChunkConstants.SIZE_X,
+                        chunkPosition.y * ChunkConstants.SIZE_Y,
+                        chunkPosition.z * ChunkConstants.SIZE_Z,
+                        true);
+                chunkShader.setFloat("animated", chunk.isAnimated() ? 1.0f : 0.0f, true);
+
+                // Effectively this just positions the chunk appropriately, relative to the camera.
+                // chunkPositionRelativeToCamera = chunkCoordinates * chunkDimensions - cameraCoordinate
+                GL11.glPushMatrix();
+                GL11.glTranslatef(chunkPositionRelativeToCamera.x, chunkPositionRelativeToCamera.y, chunkPositionRelativeToCamera.z);
+
+                chunk.getMesh().render(ChunkMesh.RenderPhase.ALPHA_REJECT); // TODO: remove face culling from this method
+                numberOfRenderedTriangles += chunk.getMesh().triangleCount();
+
+                GL11.glPopMatrix(); // Resets the matrix stack after the rendering of a chunk.
+
```

Comment:
The content of the IF block above should theoretically be wrapped by a couple of methods, one sertting the chunkShader parameters and the other translating the chunk and rendering it. This would allow for the same code to be reused in other nodes that have identical code. 

The problem is, where to put those two methods?

Some of the affected nodes inherit from WireframeCapableNode, but another inherit from ConditionDependentNode and one more from AbstractNode.  It looks like a case for multiple inheritance but by my understanding it means there is an architectural issue to be resolved.

Suggestions?